app/transformers/PAYETransformer.scala

```diff
@@ -0,0 +1,300 @@
+          appendAttribute(
+            __ \ 'income_tax \ 'payload,
+            middleTierAmountJson("higher_rate_income_tax", higherRateTax.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'rates,
+            middleTierRateJson("higher_rate_income_tax_rate", higherRate.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'payload,
+            middleTierAmountJson("ordinary_rate_amount", dividendLowRateTaxAmount.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'payload,
+            middleTierAmountJson("ordinary_rate", dividendLowRateTax.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'rates,
+            middleTierRateJson("ordinary_rate_tax_rate", dividendLowRate.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'payload,
+            middleTierAmountJson("upper_rate_amount", dividendHigherRateTaxAmount.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'payload,
+            middleTierAmountJson("upper_rate", dividendHigherRateTax.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'rates,
+            middleTierRateJson("upper_rate_rate", dividendHigherRate.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'payload,
+            middleTierAmountJson("married_couples_allowance_adjustment", marriedCouplesAllowanceAdjust.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'payload,
+            middleTierAmountJson("marriage_allowance_received_amount", marriedCouplesAllowanceReceived.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'payload,
+            middleTierAmountJson("less_tax_adjustment_previous_year", lessTaxAdjustPreviousYear.getOrElse(0))) andThen
+          appendAttribute(
+            __ \ 'income_tax \ 'payload,
+            middleTierAmountJson("tax_underpaid_previous_year", taxUnderpaidPreviousYear.getOrElse(0))) andThen
+          appendAttribute(__ \ 'income_tax \ 'payload, middleTierAmountJson("total_income_tax", incomeTax.getOrElse(0)))
+
+      safeTransform(jsonTransformer)
+    }
+
+    def transformGovSpendingData(source: JsObject): JsObject = {
+      import play.api.libs.json.Reads._
+
+      val totalTaxAmount: Option[Double] = pickAmount(__ \ 'calculatedTotals \ 'totalIncomeTaxNics, source)
+      val taxYear = original.transform((__ \ 'taxYear).json.pick[JsNumber]).asOpt.map(n => n.as[Int])
+      val govSpendingJson = Json
+        .toJson(
+          new GovSpendingDataTransformer(Amount(BigDecimal(totalTaxAmount.getOrElse(0.0)), "GBP"), taxYear.get).govSpendReferenceDTO)
+        .as[JsObject]
+
+      val jsonTransformer = (__ \ 'gov_spending).json.update((__).read[JsObject].map { o =>
+        o ++ govSpendingJson
+      })
+      safeTransform(jsonTransformer)
+    }
+  }
```

Comment:
Each of these could be granted type safety, not have to rely on duplication of Json logic we leverage using the Play Framework, be easier to read and maintain, and would match the patterns used in the other transformers in this project.